// Copyright 2025 Cisco Systems, Inc. and its affiliates
// SPDX-License-Identifier: Apache-2.0

// This file was autogenerated by go-to-protobuf. Do not edit it manually!

syntax = "proto3";

package agntcy.identity.service.v1alpha1;

import "google/api/field_behavior.proto";
import "google/protobuf/timestamp.proto";

// Package-wide variables from generator "generated".
option go_package = "github.com/agntcy/identity-service/api/server/agntcy/identity/service/v1alpha1;identity_service_sdk_go";

// Identity Service App.
message App {
  // A unique identifier for the App.
  optional string id = 1 [(.google.api.field_behavior) = OUTPUT_ONLY];

  // A human-readable name for the App.
  optional string name = 2 [(.google.api.field_behavior) = REQUIRED];

  // A human-readable description for the App.
  optional string description = 3 [(.google.api.field_behavior) = OPTIONAL];

  // The type of the App.
  optional AppType type = 4 [(.google.api.field_behavior) = REQUIRED];

  // The DID value
  optional string resolver_metadata_id = 5 [(.google.api.field_behavior) = OUTPUT_ONLY];

  // The API Key Secret for the App.
  optional string api_key = 6 [(.google.api.field_behavior) = OUTPUT_ONLY];

  // The status of the App
  optional AppStatus status = 7 [(.google.api.field_behavior) = OUTPUT_ONLY];

  // CreatedAt records the timestamp of when the App was initially created
  optional .google.protobuf.Timestamp created_at = 8 [(.google.api.field_behavior) = OUTPUT_ONLY];
}

enum AppStatus {
  // Unspecified status
  APP_STATUS_UNSPECIFIED = 0;
  // The App has at least one active badge
  APP_STATUS_ACTIVE = 1;
  // The App has no badges
  APP_STATUS_PENDING = 2;
  // The App has all the badges revoked
  APP_STATUS_REVOKED = 3;
}

// App Type
enum AppType {
  // Unspecified Envelope Type.
  APP_TYPE_UNSPECIFIED = 0;
  // Agent A2A App Type.
  APP_TYPE_AGENT_A2A = 1;
  // Agent OASF App Type.
  APP_TYPE_AGENT_OASF = 2;
  // Agent MCP Server App Type.
  APP_TYPE_MCP_SERVER = 3;
}
